### PR TITLE
[docker] Add support for pytorch-built-in docker image

### PR DESCRIPTION
**Pull request by zrphercule**
I will add a new docker image with a pytorch built in; before that I would like to have this feature in our docker image control system.

### CODE REVIEW

File: docker/caffe2/jenkins/common/install_pytorch.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+set -ex
+
+pip install numpy
+git clone https://github.com/pytorch/pytorch.git --recursive
+cd pytorch
+git checkout 7fcfed19e7c4805405f3bec311fc056803ca7afb
```
Comment (ezyang) [inline docker/caffe2/jenkins/common/install_pytorch.sh]: Life will probably be better for you if this hash is an argument (somehow). It probably shouldn't be in the image name itself but maybe be fed in as an environment variable from the Jenkins script.

File: docker/caffe2/jenkins/common/install_pytorch.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+pip install -r requirements.txt
+/usr/bin/python3.6 setup.py install
```
Comment (ezyang) [inline docker/caffe2/jenkins/common/install_pytorch.sh]: Can you use `python` here? If you configured Python before this step, you should have the correct Python binary as part of your default PATH.

**Comment by ezyang** [comment]
Re testing, can you show that all of the existing images rebuild cleanly with this change?

**Comment by zrphercule** [comment]
This is closed since we found a way to bypass the problem without upload new docker images.

**Comment by zrphercule** [comment]
Thanks for reviewing this Ed!

**Pull request closed without merging**